include/openssl/ssl.h

```diff
@@ -1558,6 +1558,27 @@ __owur char *SSL_get_srp_username(SSL *s);
 __owur char *SSL_get_srp_userinfo(SSL *s);
 # endif
 
+/*
+ * Early callback and helpers.
+ */
+typedef struct clienthello_msg_st SSL_EARLY;
```

Comment:
Any suggestions?  Alternately we can embed clienthello_msg* and SSL* into a new ssl_early_st, as mentioned above.  Might be worth the extra type...